src/Extensions/Compatibility.UlsLoggerAdapter/UlsLogAddapter.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Omex.System.Logging;
+
+namespace Microsoft.Omex.Extensions.Compatibility.UlsLoggerAdapter
+{
+	/// <summary>
+	/// Hosted service subscribes to log events from Microsoft.Omex.System.UlsLogger and propagates them to ILogger
+	/// </summary>
+	internal class UlsLogAddapter : IHostedService
+	{
+		private static readonly Func<string, Exception, string> s_formatter = (message, exception) => message; // emulating behavior of standard formatter to avoid duplicating exception stack trace
+		private readonly ILoggerFactory m_loggerFactory;
+		private readonly ILogger<UlsLogAddapter> m_logger;
+		private readonly ConcurrentDictionary<string, ILogger> m_loggersDictionary;
+
+		public UlsLogAddapter(ILoggerFactory loggerFactory, ILogger<UlsLogAddapter> logger)
+		{
+			m_loggerFactory = loggerFactory;
+			m_logger = logger;
+			m_loggersDictionary = new ConcurrentDictionary<string, ILogger>(StringComparer.OrdinalIgnoreCase);
+		}
+
+		public Task StartAsync(CancellationToken cancellationToken)
+		{
+			ULSLogging.LogEvent += LogEvent;
+			return Task.CompletedTask;
+		}
+
+		public Task StopAsync(CancellationToken cancellationToken)
+		{
+			ULSLogging.LogEvent -= LogEvent;
+			return Task.CompletedTask;
+		}
+
+		private void LogEvent(object sender, LogEventArgs e)
+		{
+			m_loggersDictionary.GetOrAdd(e.CategoryId.Name, m_loggerFactory.CreateLogger).Log(
+				Convert(e.Level),
+				(int)e.TagId,
+				e.FullMessage,
+				e is ReportExceptionEventArgs errorArgs ? errorArgs.Exception : null,
+				s_formatter);
+		}
+
+		private LogLevel Convert(Level level)
+		{
+			if (Levels.Error == level)
```

Comment:
I think you should use a switch here.

Reply:
Or create a dict like:
```
var dict = new Dictionary<Levels, LogLevel>
{
    [Levels.Error] = LogLevel.Error,
    [Levels.Warning] = LogLevel.Warning,
    ...
};
return dict[level]
```

Reply:
Switch won't work since ULS levels are not constants, I'll create a dictionary

Reply:
I've actually managed to make switch working by using underlying enum, thank you for pointing it out